docs/protocol.md

```diff
@@ -204,6 +209,32 @@ Here are the different methods that can be called:
 * Heartbeat
 * GetMessagesById
 
+### Greeting a server
+🧭 **RPC Message** > **Query** > **Greet Server**
+
+By executing a greet server action, a server can send its public key as well as the 
+addresses of its client and server endpoints to another server.
+
+This action should be executed by the server that opens a connection with another server. The 
+latter should then reply with a `GreetServer` answer containing its own information.
```

Comment:
Now that I think about it, shouldn't the answer rather be in a `result`, otherwise they might start sending each other `GreetServer`'s until the end of time ?

Reply:
I might be wrong about the scala backend (so correct me if I am) but on the go side the implementation would be much easier if the answer was a query because adding a new type of `result` would be a mess since it expects an array of messages or a map between a channel and an array of messages for now. We already keep track of the queries sent so when we receive one we just check if we sent a query with this `id `and either send a `greet_server `or we don't depending on that condition.

Reply:
I don't think we keep track of any queries on scala, I guess we will start now ;)

Reply:
I am a bit confused as to your reasoning here. Are you sure what you want is not a JSON-RPC Notification ?

It seems your message is not really an RPC, you're not calling a function and expecting a return value.

You do expect "an answer" (from a logic perspective), but the "connected-to" server would already know what to send in his own `GreetServer` - and could, in principle, send it of his own accord without breaking the protocol.

Basically, it's a bit like the heartbeat and get_messages_by_id you're familiar with. If you know the application protocol, you know that a heartbeat may trigger a get_messages_by_id, but one is not the answer to the other (at the JSON-RPC level), 

Also, you don't have a choice from a JSON-RPC perspective, if you do a query, you expect an answer object which contains a `result` field. Otherwise, you're breaking the JSON-RPC protocol. The good news (for both of you) is that it seems you don't really need a query at the JSON-RPC layer 😅

Reply:
I think you're right, I wasn't aware that by doing a query we were systematically expecting an `answer` object. 
If I understood properly, a JSON-RPC Notification would simply be this same schema but without an ID field correct ? In that case , I can remove it and specify in the documentation that a `GreetServer` will trigger another `GreetServer` (if we didn't already send one to that server).